mlflow/models/utils.py

```diff
@@ -223,6 +215,33 @@ def _handle_dataframe_input(df):
                     "type": example_type,
                 }
             )
+        elif isinstance(input_example, list):
+            for i, x in enumerate(input_example):
+                if isinstance(x, np.ndarray) and len(x.shape) > 1:
+                    raise TensorsNotSupportedException(f"Row '{i}' has shape {x.shape}")
+            if all(_is_scalar(x) for x in input_example):
+                _logger.warning(
+                    "We do not convert list of scalar values to pandas DataFrame. "
+                    "If you expect to use pandas DataFrame, please construct it and pass "
+                    "to input_example instead."
+                )
```

Comment:
Not sure if we need a warning here. info is probably good enough. 

```suggestion
                _logger.info(
                    "Lists of scalar values are not converted to a pandas DataFrame. "
                    "If you expect to use pandas DataFrames for inference, please construct a DataFrame and pass it "
                    "to input_example instead."
                )
```

Reply:
Can we mark all of these as "we need to review if these are annoying" when we do acceptance testing (bugbash) of the branch? We need to make sure these are 1) useful and 2) not annoying when using these APIs. If stdout / stderr becomes polluted with messages, users will just ignore everything that's printed there.